packages/generator/ni_measurement_plugin_sdk_generator/ni_measurement_plugin_client_generator/templates/measurement_plugin_client.py.mako

```diff
@@ -0,0 +1,89 @@
+<%page args="measure_docstring, configuration_metadata, output_metadata, service_class, configuration_parameters_with_type_and_values, measure_api_parameters, output_parameters_with_type, import_modules"/>\
+\
+"""Python Measurement Plug-In Client."""
+
+from functools import cached_property
+% for module, type in import_modules.items():
+% if type == "In-Built":
+${module}
+% endif
+% endfor
+from typing import List, NamedTuple
+
+import grpc
+from _helpers import (
+    _create_file_descriptor,
+    _get_measure_request,
+    _get_resolved_service,
+)
+from ni_measurement_plugin_sdk_service._internal.parameter import decoder
+from ni_measurement_plugin_sdk_service._internal.parameter.metadata import ParameterMetadata
+from ni_measurement_plugin_sdk_service._internal.stubs.ni.measurementlink.measurement.v2 import (
+    measurement_service_pb2 as v2_measurement_service_pb2,
+    measurement_service_pb2_grpc as v2_measurement_service_pb2_grpc,
+)
+% for module, type in import_modules.items():
+% if type == "Custom":
+${module}
+% endif
+% endfor
+from ni_measurement_plugin_sdk_service.discovery import DiscoveryClient
+
+<% output_type = "None" %>\
+% if output_metadata:
+
+class Output(NamedTuple):
+    """Measurement result container."""
+
+    ${output_parameters_with_type}
+<% output_type = "Output" %>\
+% endif
+
+
+class MeasurementPlugInClient:
```

Comment:
Hardcoding this class name makes the client harder to use. If the user has multiple measurement plugin clients, they have to be careful how they import them.

```
from nidcpower_source_dc_voltage_client import MeasurementPlugInClient
from nidmm_measurement_client import MeasurementPlugInClient  # error, name already imported
```

The user can work around this with aliasing:
```
from nidcpower_source_dc_voltage_client import MeasurementPlugInClient as NIDCPowerSourceDCVoltageClient
from nidmm_measurement_client import MeasurementPlugInClient as NIDMMMeasurementClient
```
But this raises the question: why not name the client classes based on the service class in the first place?

Reply:
Sure, on considering this case, I have updated the naming
-> If the user gives module name:
      -> The given name will be used.
-> If the user didn't give module name:
     -> Service class will be taken and used.

->  Snake casing for module and Pascal casing for class name.

Reply:
ok, good.